test_release.py

```diff
@@ -149,3 +150,19 @@ def test_hitting_httpbin_org_http11(self):
 
             assert resp.status == 200
             assert resp.read()
+
+    def test_hitting_nghttp2_org_via_h2c_upgrade(self):
+        """
+        This tests our support for cleartext HTTP/1.1 -> HTTP/2 upgrade
+        against the most common open source HTTP/2 server implementation.
+        """
+        c = HTTPConnection('nghttp2.org:80')
+
+        # Make the request.
+        c.request('GET', '/')
+        response = c.get_response()
+
+        # Check that the response is OK and that we did upgrade to HTTP/2.
+        assert response.status == 200
+        assert response.read()
+        assert response.version == HTTPVersion.http20
```

Comment:
I did [change `test_integration_http11.py`](https://github.com/Lukasa/hyper/pull/313/files#diff-9e6d1edfa0b69f29474dd4ad10311941) in such a way that it fails without my fix. Do you think something else is necessary?